public/components/Flyout/flyouts/createDetector.js

```diff
@@ -0,0 +1,814 @@
+                  href={`${KIBANA_AD_PLUGIN}#/detectors/${detectorID}`}
+                  target="_blank"
+                >
+                  {'here'} <EuiIcon size="s" type="popout" />
+                </EuiLink>
+              }
+            </span>
+          </EuiTextColor>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiCallOut>
+  );
+}
+
+function informationCallOut(callOut) {
+  if (callOut === 'filterQueryTooSparse') {
+    return warningCallOut(DETECTOR_CREATION_CALLOUTS.FILTER_QUERY_NO_RESULTS, 'warning', 'help');
+  }
+  if (callOut === 'maxInterval') {
+    return warningCallOut(DETECTOR_CREATION_CALLOUTS.MAX_INTERVAL, 'warning', 'help');
+  }
+  if (callOut === 'notValid') {
+    return warningCallOut(DETECTOR_CREATION_CALLOUTS.NOT_VALID, 'warning', 'help');
+  } else if (callOut === 'valid') {
+    return warningCallOut(DETECTOR_CREATION_CALLOUTS.VALID, 'primary', 'check');
+  }
+  return null;
+}
+
+async function validateDetector(newValue, context) {
+  const search = {
+    searchType: 'graph',
+    timeField: newValue.time_field,
+    where: {
+      fieldName: newValue.where.fieldName === undefined ? [] : newValue.where.fieldName,
+      fieldRangeEnd: newValue.where.fieldRangeEnd === undefined ? 0 : newValue.where.fieldRangeEnd,
+      fieldRangeStart:
+        newValue.where.fieldRangeStart === undefined ? 0 : newValue.where.fieldRangeStart,
+      fieldValue: newValue.where.fieldValue === undefined ? '' : newValue.where.fieldValue,
+      operator: newValue.where.operator === undefined ? 'is' : newValue.where.operator,
+    },
+  };
+  let filterQuery = formikToWhereClause(search);
+  let adFilterQuery = '';
+  if (filterQuery) {
+    adFilterQuery = { bool: { filter: [filterQuery] } };
+  } else {
+    adFilterQuery = { match_all: { boost: 1.0 } };
+  }
+  context.adConfigs.filter_query = adFilterQuery;
+  context.adConfigs.name = newValue.name;
+  context.adConfigs.description = newValue.description;
+  context.adConfigs.time_field = newValue.time_field;
+  context.adConfigs.inidices = newValue.indices;
+  context.adConfigs.window_delay = { period: { interval: newValue.window_delay, unit: 'MINUTES' } };
+  context.adConfigs.detection_interval = {
+    period: { interval: newValue.detection_interval, unit: 'MINUTES' },
+  };
+  context.queriesForOverview.where = search.where;
+  context.queriesForOverview.filter_query = adFilterQuery;
+  const configs = context.adConfigs;
+  const httpClient = context.httpClient;
+  try {
+    const response = await httpClient.post('../api/alerting/detectors/_validate', {
+      configs,
+    });
+    let resp = _.get(response, 'data.response');
+    const {
+      data: {
+        ok,
+        response: { _id },
+      },
+    } = response;
+    if (ok) {
+      context.setFlyout(null);
+      context.renderFlyout(configs, resp, context.queriesForOverview);
+    }
+  } catch (err) {
+    if (typeof err === 'string') throw err;
```

Comment:
If error is not undefined, we should log error and throw our custom error message with line 360?